.github/workflows/build.yml

```diff
@@ -0,0 +1,44 @@
+name: Build OpenAPI docs
+on:
+  - push
+  # - pull_request
+jobs:
+  docs-build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v4
+        with:
+          node-version: 14.x
+          # Comes with npm 6. For newer Node, encountered: https://github.com/npm/cli/issues/3359
+      - run: npm install -g @redocly/openapi-cli && npm install -g redoc-cli
+      - run: npm install -g gh-openapi-docs
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - run: gh-openapi-docs
+      - name: Deploy 🚀
+        uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          branch: gh-pages
+          folder: .
+  deploy:
```

Comment:
Could we somehow avoid the redundancy with `docs-build`? Seems like only two parameters are added (`target-folder` and `clean`). Perhaps we could check the branch and operation we are on and make sure that we set these accordingly?

Alternatively, we could maybe negate the `if: github.ref == 'refs/heads/master'` in `docs-build`, so that we aren't running two slightly different deployment jobs one after another after merging into `master`?

Reply:
Ya i think this can be simplified. Let me see if i can fix this.